nx/lib/nx/binary_backend.ex

```diff
@@ -1726,6 +1726,55 @@ defmodule Nx.BinaryBackend do
     from_binary(out, result_data)
   end
 
+  @impl true
+  def take_along_axis(
+        %T{type: output_type} = output,
+        %T{shape: t_shape, type: {_, t_size} = t_type} = tensor,
+        %T{shape: idx_shape, type: {_, idx_size} = idx_type} = indices,
+        axis
+      ) do
+    permute = fn t ->
+      permutation =
+        t
+        |> Nx.axes()
+        |> List.delete(axis)
+        |> List.insert_at(Nx.rank(t) - 1, axis)
```

Comment:
Maybe you want to use List.update_at?

Reply:
This effectively shifts the list left, so I don't see how update_at could be applied easily

```elixir
iex(1)> [1, 2, 3, 4] |> List.delete(2) |> List.insert_at(4 - 1, 2)
[1, 3, 4, 2]
```